DriverDataUI/src/main/java/com/drivequant/drivekit/ui/extension/TripExtension.kt

```diff
@@ -13,50 +13,26 @@ import com.drivequant.drivekit.common.ui.extension.formatDateWithPattern
 import com.drivequant.drivekit.common.ui.extension.resSpans
 import com.drivequant.drivekit.common.ui.utils.DKDatePattern
 import com.drivequant.drivekit.common.ui.utils.DKSpannable
+import com.drivequant.drivekit.core.scoreslevels.DKScoreType
 import com.drivequant.drivekit.databaseutils.entity.Trip
 import com.drivequant.drivekit.ui.DriverDataUI
 import com.drivequant.drivekit.ui.tripdetail.activity.TripDetailActivity
 import java.text.SimpleDateFormat
 import java.util.Date
 import java.util.Locale
 
-fun List<Trip>.computeSafetyScoreAverage(): Double {
-    val scoredTrips = this.filter { it.safety?.safetyScore != null && it.safety?.safetyScore!! <= 10.0 }
-    return if (this.isEmpty() || scoredTrips.isEmpty()){
+fun List<Trip>.computeAverageScore(scoreType: DKScoreType): Double =
+    if (this.isEmpty()) {
         11.0
     } else {
-        val sumScore = scoredTrips.mapNotNull { it.safety?.safetyScore }.sum()
-        sumScore.div(scoredTrips.size)
-    }
-}
-
-fun List<Trip>.computeEcoDrivingScoreAverage(): Double {
-    val scoredTrips = this.filter { it.ecoDriving?.score != null && it.ecoDriving?.score!! <= 10.0 }
-    return if (this.isEmpty() || scoredTrips.isEmpty()) {
-        11.0
-    } else {
-        val sumScore = scoredTrips.mapNotNull { it.ecoDriving?.score }.sum()
-        sumScore.div(scoredTrips.size)
-    }
-}
-
-fun List<Trip>.computeDistractionScoreAverage(): Double {
-    return if (this.isEmpty()) {
-        11.0
-    } else {
-        val sumScore = this.mapNotNull { it.driverDistraction?.score }.sum()
-        sumScore.div(this.size)
-    }
-}
-
-fun List<Trip>.computeSpeedingScoreAverage(): Double {
-    return if (this.isEmpty()) {
-        11.0
-    } else {
-        val sumScore = this.mapNotNull { it.speedingStatistics?.score }.sum()
-        sumScore.div(this.size)
+        val tripsWithScore = when (scoreType) {
```

Comment:
Ici, il faut d'abord récupérer les trajets qui ont un score et ensuite diviser la somme des scores par le nombre de trajets qui ont score.
Le problème avec le code actuel en production est que si des trajets ont `unscored` à `false` mais n'ont pas de score de distraction ou de speeding (par ex: le score est activé côté backend après que le trajet a été effectué), alors la division par `this.size` est incorrecte.